main.go

```diff
@@ -73,30 +82,38 @@ func main() {
 	r.Handler("GET", *metricsPath, prometheus.Handler())
 
 	// Handlers for pushing and deleting metrics.
-	r.PUT("/metrics/job/:job/*labels", handler.Push(ms, true))
-	r.POST("/metrics/job/:job/*labels", handler.Push(ms, false))
-	r.DELETE("/metrics/job/:job/*labels", handler.Delete(ms))
-	r.PUT("/metrics/job/:job", handler.Push(ms, true))
-	r.POST("/metrics/job/:job", handler.Push(ms, false))
-	r.DELETE("/metrics/job/:job", handler.Delete(ms))
+	pushAPIPath := *routePrefix + "/metrics"
+	r.PUT(pushAPIPath+"/job/:job/*labels", handler.Push(ms, true))
+	r.POST(pushAPIPath+"/job/:job/*labels", handler.Push(ms, false))
+	r.DELETE(pushAPIPath+"/job/:job/*labels", handler.Delete(ms))
+	r.PUT(pushAPIPath+"/job/:job", handler.Push(ms, true))
+	r.POST(pushAPIPath+"/job/:job", handler.Push(ms, false))
+	r.DELETE(pushAPIPath+"/job/:job", handler.Delete(ms))
 
 	// Handlers for the deprecated API.
-	r.PUT("/metrics/jobs/:job/instances/:instance", handler.LegacyPush(ms, true))
-	r.POST("/metrics/jobs/:job/instances/:instance", handler.LegacyPush(ms, false))
-	r.DELETE("/metrics/jobs/:job/instances/:instance", handler.LegacyDelete(ms))
-	r.PUT("/metrics/jobs/:job", handler.LegacyPush(ms, true))
-	r.POST("/metrics/jobs/:job", handler.LegacyPush(ms, false))
-	r.DELETE("/metrics/jobs/:job", handler.LegacyDelete(ms))
-
-	r.Handler("GET", "/static/*filepath", prometheus.InstrumentHandler(
+	r.PUT(pushAPIPath+"/jobs/:job/instances/:instance", handler.LegacyPush(ms, true))
+	r.POST(pushAPIPath+"/jobs/:job/instances/:instance", handler.LegacyPush(ms, false))
+	r.DELETE(pushAPIPath+"/jobs/:job/instances/:instance", handler.LegacyDelete(ms))
+	r.PUT(pushAPIPath+"/jobs/:job", handler.LegacyPush(ms, true))
+	r.POST(pushAPIPath+"/jobs/:job", handler.LegacyPush(ms, false))
+	r.DELETE(pushAPIPath+"/jobs/:job", handler.LegacyDelete(ms))
+
+	r.Handler("GET", *routePrefix+"/static/*filepath", prometheus.InstrumentHandler(
 		"static",
 		http.FileServer(
-			&assetfs.AssetFS{Asset: Asset, AssetDir: AssetDir, AssetInfo: AssetInfo},
+			&assetfs.AssetFS{
+				Asset: func(name string) ([]byte, error) {
+					// Note: this will do bad things if routePrefix is multi-byte,
+					// but I don't think that's valid in a URL anyway.
+					path := name[len(*routePrefix):]
+					return Asset(path)
+				},
+				AssetDir: AssetDir, AssetInfo: AssetInfo},
 		),
 	))
 	statusHandler := prometheus.InstrumentHandlerFunc("status", handler.Status(ms, Asset, flags))
-	r.Handler("GET", "/status", statusHandler)
-	r.Handler("GET", "/", statusHandler)
+	r.Handler("GET", *routePrefix+"/status", statusHandler)
+	r.Handler("GET", *routePrefix+"/", statusHandler)
 
 	// Re-enable pprof.
 	r.GET("/debug/pprof/*pprof", handlePprof)
```

Comment:
We probably want the routing prefix here as well?

Reply:
My thought with `/debug` is that it's not necessarily something you want to expose to the same audience as the other routes. I suppose this is something that can be blocked by reverse proxies, though.

I'll prefix this for now.